ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/DepositsValidator.java

```diff
@@ -64,10 +77,47 @@ public boolean validateDepositsRoot(final Block block) {
   class AllowedDeposits implements DepositsValidator {
 
     private static final Logger LOG = LoggerFactory.getLogger(AllowedDeposits.class);
+    private final Address depositContractAddress;
+
+    public AllowedDeposits(final Address depositContractAddress) {
+      this.depositContractAddress = depositContractAddress;
+    }
+
+    public Address getDepositContractAddress() {
+      return depositContractAddress;
+    }
+
+    @Override
+    public boolean validateDepositParameter(final Optional<List<Deposit>> deposits) {
+      return deposits.isPresent();
+    }
 
     @Override
-    public boolean validateDeposits(final Optional<List<Deposit>> deposits) {
-      return true; // TODO 6110: Validate deposits in the next PR
+    public boolean validateDeposits(final Block block, final List<TransactionReceipt> receipts) {
+      if (block.getBody().getDeposits().isEmpty()) {
+        LOG.warn("Deposits must not be empty when Deposits are activated");
+        return false;
+      }
+
+      List<Deposit> actualDeposits = new ArrayList<>(block.getBody().getDeposits().get());
+      List<Deposit> expectedDeposits = new ArrayList<>();
+
+      for (TransactionReceipt receipt : receipts) {
+        for (Log log : receipt.getLogsList()) {
+          if (depositContractAddress.equals(log.getLogger())) {
+            Deposit deposit = DepositDecoder.decodeFromLog(log);
+            expectedDeposits.add(deposit);
+          }
+        }
+      }
+
+      boolean isValid = actualDeposits.equals(expectedDeposits);
+
+      if (!isValid) {
+        LOG.warn("Deposits validation failed. Deposits from block body do not match deposits from logs");
```

Comment:
Would be great to add at least the block hash here so we can tell where the problem was, unless that would already happen in the level above?

You could also add the actual vs expected deposits to make debugging easier, but probably would need to go at DEBUG or TRACE level in a second log message, it would be too noisy for WARN.